src/batch/src/executor/utils.rs

```diff
@@ -0,0 +1,31 @@
+use futures::stream::BoxStream;
+use futures::StreamExt;
+use futures_async_stream::try_stream;
+use risingwave_common::array::DataChunk;
+use risingwave_common::error::{Result, RwError};
+
+use crate::executor::BoxedDataChunkStream;
+
+pub type BoxedDataChunkListStream = BoxStream<'static, Result<Vec<DataChunk>>>;
+
+/// Read at least `rows` rows.
+#[try_stream(boxed, ok = Vec<DataChunk>, error = RwError)]
+pub async fn batch_read(mut stream: BoxedDataChunkStream, rows: usize) {
+    let mut cnt = 0;
+    let mut chunk_list = vec![];
+    while let Some(build_chunk) = stream.next().await {
+        let build_chunk = build_chunk?;
+        cnt += build_chunk.cardinality();
+        chunk_list.push(build_chunk);
+        if cnt < rows {
+            continue;
+        } else {
+            yield chunk_list;
+            cnt = 0;
+            chunk_list = vec![];
+        }
+    }
+    if !chunk_list.is_empty() {
+        yield chunk_list;
+    }
```

Comment:
> LGTM!
> 
> Actually I'm not sure how much will improve. I remembered in the early impl of batch HashJoin we just read all chunks of build side into memory...

Early batch lookup join I think? for query likes `select * from sbtest1 where k in (1,2,3,4)`, it might read 4 small chunks and this PR can merge them and just lookup once.

Reply:
> What about always yield?

It seems make no sense to return an empty chunk list.